sonar-scanner-engine/src/main/java/org/sonar/scanner/issue/IssueTransformer.java

```diff
@@ -52,7 +52,8 @@ public static TrackedIssue toTrackedIssue(ServerIssue serverIssue) {
     issue.setEndLine(serverIssue.hasLine() ? serverIssue.getLine() : null);
     issue.setSeverity(serverIssue.getSeverity().name());
     issue.setAssignee(serverIssue.hasAssigneeLogin() ? serverIssue.getAssigneeLogin() : null);
-    issue.setComponentKey(ComponentKeys.createEffectiveKey(serverIssue.getModuleKey(), serverIssue.hasPath() ? serverIssue.getPath() : null));
+    String keyWithoutBranch = StringUtils.substringBefore(serverIssue.getModuleKey(), ":");
```

Comment:
In fact I'm wrong. Just by looking at a remote project key A:B you can't tell if B is a branch or not.